src/object/obj_tx.c

```diff
@@ -1167,6 +1895,7 @@ dc_tx_restart(tse_task_t *task)
 
 		tx->tx_status = TX_OPEN;
 		tx->tx_epoch.oe_value = 0;
+		tx->tx_epoch.oe_flags |= DTX_EPOCH_UNCERTAIN;
```

Comment:
Restarting a TX should not change its epoch uncertainty on the client side. This change should be unnecessary.

Reply:
Since the TX needs to be restarted, then it will use new epoch, since we do not know which epoch will be used, then it is "DTX_EPOCH_UNCERTAIN", right?

Reply:
If UNCERTAIN is not set originally, it shall remain unset after a restart; if it is set originally, it shall remain set after a restart. So, restarting a TX doesn't need to set or clear UNCERTAIN.